x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/RowInTableOperator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.operator;
+
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.compute.aggregation.table.RowInTable;
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.core.ReleasableIterator;
+import org.elasticsearch.core.Releasables;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class RowInTableOperator extends AbstractPageMappingToIteratorOperator {
+    public record Key(String name, Block block) {
+        @Override
+        public String toString() {
+            return "{name="
+                + name
+                + ", type="
+                + block.elementType()
+                + ", positions="
+                + block.getPositionCount()
+                + ", size="
+                + ByteSizeValue.ofBytes(block.ramBytesUsed())
+                + "}";
+        }
+    }
+
+    /**
+     * Factory for {@link RowInTableOperator}. It's received {@link Block}s
+     * are never closed, so we need to build them from a non-tracking factory.
+     */
+    public record Factory(Key[] keys, int[] blockMapping) implements Operator.OperatorFactory {
+        @Override
+        public Operator get(DriverContext driverContext) {
+            return new RowInTableOperator(driverContext.blockFactory(), keys, blockMapping);
+        }
+
+        @Override
+        public String describe() {
+            return "RowInTable[keys=" + Arrays.toString(keys) + ", mapping=" + Arrays.toString(blockMapping) + "]";
+        }
+    }
+
+    private final List<String> keys;
+    private final RowInTable lookup;
```

Comment:
should we call this `rowInTable` or `table` instead?

Reply:
++ - old names didn't get changed.